rlwe/rlwe_test.go

```diff
@@ -537,6 +538,44 @@ func testMarshaller(kgen KeyGenerator, t *testing.T) {
 		assert.True(t, params.Equals(rlweParams))
 	})
 
+	t.Run(testString(params, "Marshaller/Ciphertext/EndToEnd/"), func(t *testing.T) {
+
+		prng, _ := utils.NewPRNG()
+
+		ciphertextWant := NewCiphertextRandom(prng, params, 2, params.MaxLevel())
+
+		marshalledCiphertext, err := ciphertextWant.MarshalBinary()
+		require.NoError(t, err)
+
+		ciphertextTest := new(Ciphertext)
+		require.NoError(t, ciphertextTest.UnmarshalBinary(marshalledCiphertext))
+
+		require.Equal(t, ciphertextWant.Degree(), ciphertextTest.Degree())
+		require.Equal(t, ciphertextWant.Level(), ciphertextTest.Level())
+
+		for i := range ciphertextWant.Value {
+			require.True(t, params.RingQ().EqualLvl(ciphertextWant.Level(), ciphertextWant.Value[i], ciphertextTest.Value[i]))
+		}
+	})
+
+	t.Run(testString(params, "Marshaller/Ciphertext/Minimal/"), func(t *testing.T) {
```

Comment:
I don't get the point of separating in two tests cases? Can the two be merged in one, potentially with an internal loop repeating the test for several ct degree ?.

Reply:
I suppose, only one test with degree 2 would be sufficient

Reply:
Added the test for degree 0, 1, 2, 3, 4 for future-proofness.